src/haz3lcore/statics/TypBase.re

```diff
@@ -244,18 +275,17 @@ module rec Typ: {
       !resolve && eq(ty_name, ty_join) ? Var(name) : ty_join;
     /* Note: Ordering of Unknown, Var, and Rec above is load-bearing! */
     | (Rec(x1, ty1), Rec(x2, ty2)) =>
-      /* TODO:
```

Comment:
did this TODO get resolved? still looks like we're doing substitution with an open term, `Var(x1)`...

Reply:
I think this is resolved since subst() itself performs capture-avoiding substitution. I'll note that in a comment and replace this one.